be/src/exec/hash_join_components.cpp

```diff
@@ -87,8 +87,11 @@ Status HashJoinBuilder::append_chunk(RuntimeState* state, const ChunkPtr& chunk)
 }
 
 Status HashJoinBuilder::build(RuntimeState* state) {
-    SCOPED_TIMER(_hash_joiner.build_metrics().build_ht_timer);
-    TRY_CATCH_BAD_ALLOC(RETURN_IF_ERROR(_ht.build(state)));
+    // if query is already cancelled, building hash table is unnecessary, just skip it
+    if (!state->is_cancelled()) {
+        SCOPED_TIMER(_hash_joiner.build_metrics().build_ht_timer);
+        TRY_CATCH_BAD_ALLOC(RETURN_IF_ERROR(_ht.build(state)));
+    }
     _ready = true;
     return Status::OK();
```

Comment:
still return ok if cancelled?

Reply:
> still return ok if cancelled?

yeah, my idea is to skip this process only, leaving an empty hash table here, and keep other processes the same as before, so the risk is relatively small.

Reply:
> > still return ok if cancelled?
> 
> yeah, my idea is to skip this process only, leaving an empty hash table here, and keep other processes the same as before, so the risk introduced by my change is relatively small.